concent_api/core/views.py

```diff
@@ -538,6 +663,14 @@ def store_message_and_message_status(golem_message_type: int, task_id: str, raw_
     golem_message.full_clean()
     golem_message.save()
 
+    message_auth = MessageAuth(
+        message                    = golem_message,
+        provider_public_key_bytes  = provider_public_key,
+        requestor_public_key_bytes = requestor_public_key,
```

Comment:
You're creating `MessageAuth` for every message. Why create a separate table then? Either reuse same requestor/provider key object for all messages in a single task or just move these fields to `Message`.

Reply:
I have asked you about this before I have started working on this branch. You have said two things - store separate object for each message and make it a separate model. Why do you want to change it now, at the end of work ?

Reply:
I'm not sure how exactly I worded it but it looks like a misunderstanding. The whole purpose of creating a new model is to be able to create a relationship that's not 1:1. Otherwise we would not need a separate model at all.

We can leave it out of this pull request but it's really something that can't stay that way in the long term. I'm creating another issue for it.

Reply:
:+1: Lets see how next level of authentication will be described.